gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/TimeAwareRecursiveCopyableDataset.java

```diff
@@ -134,9 +134,51 @@ protected List<FileStatus> getFilesAtPath(FileSystem fs, Path path, PathFilter f
     return recursivelyGetFilesAtDatePath(fs, path, "", fileFilter, 1, startDate, endDate, formatter);
   }
 
+  /**
+   * Checks if the datePath provided is in the range of the start and end dates.
+   * Rounds startDate and endDate to the same granularity as datePath prior to comparing.
+   * Returns true if the datePath provided is in the range of start and end dates, inclusive.
+   * @param startDate
+   * @param endDate
+   * @param datePath
+   * @param datePathFormat (This is the user set desired format)
+   * @param level
+   * @return true/false
+   */
+  public Boolean checkPathDateTimeValidity(LocalDateTime startDate, LocalDateTime endDate, String datePath, String datePathFormat, int level) {
+    String [] array = datePathFormat.split("/");
+    StringBuilder datePathPattern = new StringBuilder();
+
+    for (int index = 1; index < level; index++) {
+      if (index > 1) {
+        datePathPattern.append("/");
+      }
+      datePathPattern.append(array[index - 1]);
+    }
+
+    try {
+      DateTimeFormatter formatGranularity = DateTimeFormat.forPattern(datePathPattern.toString());
+      LocalDateTime traversedDatePathRound = formatGranularity.parseLocalDateTime(datePath);
+      LocalDateTime startDateRound = formatGranularity.parseLocalDateTime(startDate.toString(datePathPattern.toString()));
+      LocalDateTime endDateRound = formatGranularity.parseLocalDateTime(endDate.toString(datePathPattern.toString()));
+
+      boolean afterOrOnStartDate = traversedDatePathRound.isAfter(startDateRound) || traversedDatePathRound.isEqual(startDateRound);
+      boolean beforeOrOnEndDate = traversedDatePathRound.isBefore(endDateRound) || traversedDatePathRound.isEqual(endDateRound);
+      return afterOrOnStartDate && beforeOrOnEndDate;
```

Comment:
You can probably simplify this by
`return !traversedDatePathRound.isBefore(startDateRound) && !traversedDatePathRound.isAfter(endDateRound)`